expression/builtin_vectorized_test.go

```diff
@@ -783,6 +785,36 @@ func (s *testEvaluatorSuite) TestFloat32ColVec(c *C) {
 	c.Assert(col.VecEvalReal(ctx, chk, result), IsNil)
 }
 
+func (s *testEvaluatorSuite) TestSleepVec(c *C) {
+	ctx := mock.NewContext()
+	sessVars := ctx.GetSessionVars()
+
+	fc := funcs[ast.Sleep]
+	// non-strict model
+	sessVars.StrictSQLMode = false
+	d := make([]types.Datum, 1)
+	f, err := fc.getFunction(ctx, s.datumsToConstants(d))
+	c.Assert(err, IsNil)
+
+	start := time.Now()
+	go func() {
+		time.Sleep(1 * time.Second)
+		atomic.CompareAndSwapUint32(&ctx.GetSessionVars().Killed, 0, 1)
+	}()
+
+	plus, input, buf := genMockVecPlusIntBuiltinFunc()
```

Comment:
Could you make your own input here but not use `genMockVecPlusIntBuiltinFunc`?
And in `genMockVecPlusIntBuiltinFunc`, it will sleep 1 seconds totally, which may not meet `ctx.GetSessionVars().Killed`.